storage/boltz/query_bolt_cursors.go

```diff
@@ -77,6 +77,13 @@ func (f *ReverseBoltCursor) Next() error {
 	return nil
 }
 
+func NewTypedBoltCursor(cursor *bbolt.Cursor, forward bool) ast.SetCursor {
+	if forward {
```

Comment:
If the caller knows what to use for forward why don't they just instantiate the concrete the actually want?

I am not a fan of flags like this. Usually, they signify that you have two separate methods, types, etc.

Reply:
Ok, I inlined it.